### ISSUE TITLE
Path does not load on mini-map after a jump

### ISSUE THREAD

**Issue by adash12** [ProjectSidewalk/SidewalkWebpage#776]
![image](https://user-images.githubusercontent.com/13080218/27642725-a727ed20-5bed-11e7-9219-5f69b819b4a9.png)
Occasionally after a manual jump, the compass will point the user in the direction of a path that hasn't appeared on the mini-map yet. Taking a step in that direction makes the path appear.

![image](https://user-images.githubusercontent.com/13080218/27642797-d6195060-5bed-11e7-9e6a-5fed5e8ea3f1.png)

**Comment by misaugstad** [comment]
To me, it looks like you started out facing in the correct direction, with the first audit task having been loaded, it was just a very short audit task. Then after walking one step, that tiny audit was marked as complete and a new one came up, and you were facing in the correct direction.

It looks like the compass was always in the correct direction, you just got dropped not exactly on the start point of the first audit task... And since that audit task is so small, you essentially were dropped at the end of it so it looks like you are facing the wrong way. 

Does that sound right to others?

**Comment by manaswis** [comment]
Yes, that's exactly what I thought as well. Its because of the task/street being super small.

**Comment by jonfroehlich** [comment]
I personally think this reflects a larger problem, which is that a mission should just load the entire route when it starts. None of this weird task/street stuff (which is an internal data structure anyway and the user doesn't care about it). And loading the entire route on the backend first would allow us to vet and process it and make sure it's reasonable...

**Comment by jonfroehlich** [comment]
Closing after a code + git-history review (part of the pre-2020 triage in #4372) — both halves of this issue have been resolved by later work:

**The reported symptom can no longer occur.** All streets in the region (or route) are now fetched in a single request at page load (`TaskContainer.fetchTasks()` → `/tasks?regionId=` or `/routeTasks?userRouteId=`) and *all* of them are drawn on the minimap at page load (`renderAllTasks()`). When the next street is selected, it renders synchronously before the compass updates — there's no server round-trip in between, so the compass can't point at a not-yet-drawn street. The tiny-street root cause @misaugstad diagnosed is also mitigated: `nextTask()` now explicitly sets the start point/direction of the new street, and jumps go through the deliberate "label before jump" flow.

**The feature ask — "load the entire route up front, vetted on the backend" — shipped via RouteBuilder (2023).** Routes are defined server-side (`user_route`/`route_street`), loaded whole at page load, and followed in deterministic order (`?routeId=` on Explore). Default explore missions remain distance-based with priority-driven street selection, which supplies the vetting in a different form; if backend-precomputed routes for default missions are still wanted, that should be a fresh, scoped issue.

The one living remnant — tiny border street segments being awkward (now in RouteBuilder) — is tracked in #3987.
